frontend/src/pages/profile/components/edit-form/edit-form.tsx

```diff
@@ -0,0 +1,85 @@
+import { Form } from '~/libs/components/components.js';
+import { AuthMode } from '~/libs/enums/enums.js';
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { useAppSelector, useCallback } from '~/libs/hooks/hooks.js';
+import {
+  type UserEntityObjectWithGroupAndBusinessT,
+  type ValueOf,
+} from '~/libs/types/types.js';
+import {
+  type BusinessEditDto,
+  type CustomerEditDto,
+  businessEditValidationSchema,
+  customerEditValidationSchema,
+} from '~/packages/users/users.js';
+import { selectUser } from '~/slices/auth/selectors.js';
+
+import { editBusinessFields, editCustomerFields } from './libs/fields.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  onSubmit: (payload: CustomerEditDto | BusinessEditDto) => void;
+  mode: ValueOf<typeof AuthMode>;
+};
+
+const EditForm: React.FC<Properties> = ({ onSubmit, mode }: Properties) => {
+  const { firstName, lastName, phone, email, business } = useAppSelector(
+    selectUser,
+  ) as UserEntityObjectWithGroupAndBusinessT;
+
+  const getScreen = useCallback((): React.ReactNode => {
```

Comment:
Maybe put it in the hook to make the component cleaner?

Reply:
This can certainly be done, but I don't think it's necessary. This code is not reusable, it is used only in this component, and it does not overload it with code. So it's 50/50, but I would leave it as it is for now.